### PR TITLE
Sorted api resource operations

### PR DESCRIPTION
**Pull request by alexandrtspl**
* add ApiResource "operations" sorting support

https://eonx.atlassian.net/browse/PLU-5046

### CODE REVIEW

**Review by itorgov — APPROVED** [review]
__eonx_placeholder__

File: src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php [hunk 1]
```diff
@@ -11,16 +11,16 @@
...
  * within the 'ApiResource' annotation are properly sorted.
  *
- * The sniff works only with API Platform version 2.7 or earlier and checks if the operation
- * keys are sorted according to specific criteria. The sorting criteria are defined in
- * the getRanks method. If the keys are not sorted properly, the sniff will provide an
- * option to fix the order.
+ * The sorting criteria are defined in getRanks and getOperationsRanks methods.
+ * If the keys are not sorted properly, the sniff will provide an option to fix the order.
  */
 final class SortedApiResourceOperationKeysSniff implements Sniff
```
Comment (alexndlm) [inline src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php:25]: I think we should deprecate this sniff and create a new one with `SortApiResourceOperationsSniff`, which will work only with `operations`.

@roman-eonx wdyt?

**Review by alexndlm — CHANGES_REQUESTED** [review]

**Comment by alexndlm** [comment]
Please check it on PG.

File: src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php (hunk 1 above)
Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php:25]: Why? 90% of code is reused in this sniff.

Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php:25]: But I agree for us it's not useful anymore

**Comment by alexandrtspl** [comment]
> Please check it on PG.

It's checked before PR

File: src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php (hunk 1 above)
Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php:27]: Done

Comment (alexndlm) [reply src/Sniffs/Attributes/SortedApiResourceOperationKeysSniff.php:27]: Thanks

**Review by alexndlm — CHANGES_REQUESTED** [review]

File: src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+{
+    /**
+     * @var string
+     */
+    private const API_RESOURCE_OPERATIONS_NOT_SORTED = 'ApiResourceOperationsNotSorted';
+
+    /**
+     * @var array<string>
+     */
+    private const API_RESOURCE_OPERATIONS_TO_PROCESS = ['operations'];
```
[the remark this answers is not in the document]
Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php:33]: Done

File: src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\ArrayItem[] $items
+     */
+    private function isNotAssociativeOperationsArrayOnly(array $items): bool
```
[the remark this answers is not in the document]
Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php:190]: Yes. Array-style operations should be skipped anyway.

[the remark this answers is not in the document]
Comment (alexandrtspl) [reply src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php]: Removed

**Review by alexndlm — APPROVED** [review]

File: src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexndlm) [reply src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php]: thanks

File: src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (alexndlm) [reply src/Sniffs/Attributes/SortedApiResourceOperationsSniff.php]: thanks

**Pull request merged**